sklearn/pipeline.py

```diff
@@ -1224,6 +1224,7 @@ def __sklearn_tags__(self):
                 tags.input_tags.pairwise = get_tags(
                     self.steps[0][1]
                 ).input_tags.pairwise
+                tags.input_tags.sparse = get_tags(self.steps[0][1]).input_tags.sparse
```

Comment:
Ouch I don't think this is that simple. To be complete, we should check the sparse of each step and somewhow the output as well. If the previous output agree with the next input then we would be fine.

Somehow, we might want to be lenient and set it to True and hope that the underlying estimator will just raise.

I would need a second opinion here.

Reply:
Maybe @adrinjalali ?

Reply:
Yep indeed I would do something similar to what you did for the `FeatureUnion` and checking that each steps accept sparse data.

Reply:
done in a9fb7d7